### ISSUE TITLE
field.widget.format_value(cleaned_value) changes variable types when using Django forms

### ISSUE THREAD

**Issue by Tubo** [adamghill/django-unicorn#617]
If I have a form class like this:

```
class LeaveForm(forms.ModelForm):
    class Meta:
        model = Leave
        fields = ["date", "type", "portion", "comment"]
```
where "type" and "portion" are `ChoiceField`. 

I have a Unicorn view like this:
```
class LeaveFormView(UnicornView):
    form_class = LeaveForm
    date: date = None
    type: str = "" 
    portion: str = ""
    comment: str = ""
```

If I select an option with this template:

```
        <select id="inputState"
                u:model="portion">
            <option value="ALL">ALL</option>
		</select>
```

I expect `portion` to be `"ALL"`. However, I get `["ALL"]` instead. This causes validation error.

After some debugging, I discovered that this line in `unicorn_view.py` line 468 has this call:

```
value = field.widget.format_value(cleaned_value)
```

which in turn calls `ChoiceWidget.format_value` which returns a list. I presume this gets propagated to the front-end and returns as a list.

I'm not sure of how Unicorn works, but I presume potential solution to this would include:

1. Special case treatment for `ChoiceField` and `TypedChoiceField`. Both of the above uses `Select` widget which inherits from `ChoiceWidget`. You could make a special conditional for these two fields to coerce list into a basic type.
2. Alternatively, `ChoiceField` has a `allow_multiple_selected` attribute. You could use that to decide whether to keep list or not.

Thanks again for this amazing project.

-T

**Comment by adamghill** [comment]
Interesting! Thanks for tracking this down and giving some sample code. I'll create a test for this and see how easy it is to handle -- it _seems_ like it should be straight-forward to handle (I hope!).

**Comment by adamghill** [comment]
A fix for this has been merged and will be released in 0.58.0.

**Comment by adamghill** [comment]
It didn't seem like enough for a minor version bump, so I just published 0.57.1 with this fix. Let me know if you run into any problems using it!
